lib/oscap_tests.pm

```diff
@@ -123,186 +711,541 @@ sub upload_logs_reports {
+        my $exclusions_file_name = "openqa_tests_exclusions.yaml";
+        my $url = "https://gitlab.suse.de/seccert-public/compliance-as-code-compiled/-/raw/main/content/";
+
+        download_file_from_https_repo($url, $exclusions_file_name);
+        upload_logs("$exclusions_file_name") if script_run "! [[ -e $exclusions_file_name ]]";
+        my $data = script_output("cat $exclusions_file_name", quiet => 1);
+
+        # Pharse the expected results
+        my $exclusions_data = YAML::PP::Load($data);
+        record_info("Looking exclusions", "Looking exclusions for \nprofile_ID: $profile_ID\ntype: $type\narch: $arch");
+
+        $exclusions = $exclusions_data->{$profile_ID}->{$type}->{$arch}->{$exclusions_list_name};
+        # If results defined
+        if (defined $exclusions) {
+            $found = 1;
+            if (not defined $$exclusions[0]) {
+                my @exclusions = ();
+                $_[0] = \@exclusions;
+                record_info("Got exclusions", "Got exclusions for \nprofile_ID: $profile_ID\ntype: $type\narch: $arch\nList excluded rules:\n @exclusions");
+            }
+            else {
+                $_[0] = $exclusions;
+                record_info("Got exclusions", "Got exclusions for \nprofile_ID: $profile_ID\ntype: $type\narch: $arch\nList of excluded rules:\n" . (join "\n", @$exclusions));
+            }
+        }
+        else {
+            record_info("No exclusions", "No exclusions found in \nfile: $exclusions_file_name\n for profile_ID: $profile_ID\ntype: $type\narch: $arch");
+            my @exclusions = ();
+            $_[0] = \@exclusions;
+        }
+        return $found;
+    }
+}
 
+sub oscap_security_guide_setup {
+    # Main test setup function
     select_console 'root-console';
+    # Setting $full_ansible_file_path aftr got ansible_profile_ID from test
+    $full_ansible_file_path = $ansible_file_path . $ansible_profile_ID;
 
+    # Refresh repositories
+    zypper_call('ref -s', timeout => 180);
     # Install packages
     zypper_call('in openscap-utils scap-security-guide', timeout => 180);
-
     # Record the pkgs' version for reference
     my $out = script_output("zypper se -s openscap-utils scap-security-guide");
     record_info("Pkg_ver", "openscap security guide packages' version:\n $out");
-
     # Set ds file
     set_ds_file();
 
     # Check the ds file information for reference
-    my $f_ssg_ds = is_sle ? $f_ssg_sle_ds : $f_ssg_tw_ds;
-    $out = script_output("oscap info $f_ssg_ds");
+    $f_ssg_ds = is_sle ? $f_ssg_sle_ds : $f_ssg_tw_ds;
+    $out = script_output("oscap info $f_ssg_ds", quiet => 1);
     record_info("oscap info", "\"# oscap info $f_ssg_ds\" returns:\n $out");
-
     # Check the oscap version information for reference
     $out = script_output("oscap -V");
     record_info("oscap version", "\"# oscap -V\" returns:\n $out");
+
+    # Get the tests configuration file from repository and set global configuration varables
```

Comment:
I noticed multiple typos in the comments, `varables', below 'whith', 'depencies' ans so on, as mentioned in another comment, better to avoid comments when there is obvious code, they attract this kind of problems. Please give a double-check to them and remove all that you find superfluous.

Reply:
Fixed